src/WebJobs.Extensions.DurableTask/ContextInterfaces/IInterleavingContext.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// Provides functionality available to orchestration code and reentrant actor code.
+    /// </summary>
+    public interface IInterleavingContext : IDeterministicExecutionContext
```

Comment:
Yes, the name `IInterleavingContext` is not very good, I would like to change it for something less confusing. I couldn't really come up with a good one yet though.

I am not sure about `IAsynchronousExecutionContext` - it seems a bit approximate because even the actor operations can still make asynchronous calls to activities.  Also,  "asynchronous" already has a lot of meanings (e.g. Async suffix for methods returning tasks) which can be confusing.

The point of this interface is to capture functionality that is common between normal orchestrations and "actor-orchestrations", which is really what the "reentrant/always-interleave/non-locked" operations are... they are orchestrations that run inside an actor, and have a similar API to normal orchestrations, but plus the actor stuff, and minus some stuff that only makes sense for classic standalone orchestrations (get input, continue as new instance id, parent instance id, set custom status). 

Perhaps we introduce the term "actor-orchestration" (instead of calling them interleaving operations) and then use `IOrchestrationContextCommon`. Not sure.

Reply:
Yeah, I'm not sure either, so I'm inclined to just leave it as-is for now.